pages/alerting/alerting-by-example.rst

```diff
@@ -0,0 +1,168 @@
+.. _alert_by_example:
+
+Alerting by Example
+-------------------
+
+The goal of this document is to provide explanation for the configuration
+of an Event Definition with an real life example.
+
+Scenario
+^^^^^^^^
+Graylog is getting log messages for a service, where user needs to authenticate
+to log in to the service (e.g ssh, web app).
+In this scenario we want to receive an email from Graylog if a brute force attack
+is run on the authentication of the service. So if one user failed to login to a system
+10 times in one minute than we want to get an email from Graylog.
+
+Event Details
+^^^^^^^^^^^^^
+After clicking on ``Create Event Definition`` we see the Event Definition create wizard.
+We are on the first page called ``Event Details``.
+Here we enter title and description of our Event Definition. Also we define a priority.
+The priority is a tool for the user to add a classification to a event. It will be later
+displayed in the events overview as a thermometer:
+
+.. image:: /images/priority.png
+
+Since we get a message about a brute force attack which should be treated urgently we set
+the priority on ``high`` and go on to the next step.
+
+.. image:: /images/event-details.png
+
+Condition
+^^^^^^^^^
+As ``Condition Type`` we choose ``Filter & Aggregation``. Below the selection will now the
+``Filter`` section be available.
+
+Filter
+^^^^^^
+First of all we choose the ``Stream`` in which our log files are routed. If no stream for
+the web application was created, we highly recommend :ref:`to do that <how_to_create_a_stream>`.
+That way the query result will be limited to the logs of the web application and no other logs
+can influence the filter process.
+
+Now we need to filter the incoming messages so we can later count the messages
+which are matching the filter. The log message which is indicating a
+failed authentication look like this:
+
+``Login failed for user admin from ip 240.210.133.39``
+
+So we add ``"Login failed for user"`` to our query field. On the right side should
+now a preview table appear which shows the messages matching the query.
+The preview uses ``Search within the last`` as time range. If you provoke
+a log entry now, you should see at least this message after updating the
+query.
+
+Now we set ``Search within the last`` to 10 seconds and ``Execute search every``
+to 10 seconds. Now the Event engine will execute the query every 10 seconds
+for a time range of 10 seconds.
+
+.. warning:: If  ``Search within the last`` is greater than  ``Execute search every``
+  the event engine generate multiple events for the same log line. For example if you set
+  ``Search within the last`` to one minute and  ``Execute search every`` to 10 seconds
+  the engine will find the same log entry 6 times. Since the log entry is visible for one
+  minute and we search every 10 seconds.
```

Comment:
Maybe we should add the well known terms `tumbling window` and `hopping window` here.
E.g.: 
If the ``Search within the last`` period is the same as the ``Execute search every`` period,
you create a so called `Tumbling window` which is recommended for most situations.